addons/product/models/product_attribute.py

```diff
@@ -51,11 +73,14 @@ def write(self, vals):
             self.invalidate_cache()
         return res
 
-    @api.multi
-    def _get_related_product_templates(self):
-        return self.env['product.template'].with_context(active_test=False).search([
-            ('attribute_line_ids.attribute_id', 'in', self.ids),
-        ])
+    def unlink(self):
+        for pa in self:
+            if pa.product_tmpl_ids:
+                raise UserError(
+                    _("You cannot delete the attribute <strong>%s</strong> because it is used on at least one product:\n%s") %
```

Comment:
This message sound better:

```
"You cannot delete the attribute <strong>%s</strong> because it is used on this product(s):\n\n%s"
```

Reply:
The goal was to avoid having that singular/plural "(s)"  but I can double check again with the team. But then I would probably use instead "on the following products:"

Reply:
Changed to "on the following products"